starknet_py/contract.py

```diff
@@ -172,14 +179,27 @@ async def call(
         )
         return self._payload_transformer.to_python(result)
 
-    async def invoke(self, signature: Optional[Collection[int]] = None) -> InvokeResult:
+    async def invoke(
+        self,
+        signature: Optional[Collection[int]] = None,
+        max_fee: Optional[int] = None,
```

Comment:
I think we should add a warning when max_fee is not provided

Reply:
I'm not sure how do it though. If you obtain `PreparedFunctionCall` from `ContractFunction.prepare`, you may have already provided max_fee then. There's no easy way to detect that currently since we're defaulting do `max_fee=0` when no is provided in prepare.

Reply:
We can default in the exact place where max_fee is required. All calls leading to this can use None as the default.

Reply:
Okay, how about `self.max_fee` in PreparedFunctionCall. If we start defaulting to `None` instead of `0`, it will cause problems in methods that use max_fee (e.g. `hash` or `_make_invoke_function`. 
Is defaulting to `0` in these methods okay then (for example in hash `max_fee=self.max_fee or 0`)?